_posts/2025-06-09-roadmap-update-html-report.md

```diff
@@ -0,0 +1,91 @@
+
+## TL;DR
+
+Native HTML report generation is coming in V3! This will work out of the box, without additional conversion tools, and will be free and open source!
+
+## V2 was about test execution
+
+The V2 versions of Reqnroll focused on stabilizing the Reqnroll codebase and the test execution workflow. The SpecFlow codebase [was forked at version v4-beta](2024-02-08-from-specflow-to-reqnroll-why-and-how.md). That version was never officially released and therefore was only used by a few SpecFlow projects. It contained many changes, especially related to asynchronous execution of tests. This was a significant change affecting many aspects of execution, and because it was only tried by a limited set of users, it was not "battle tested." We could say that Reqnroll V2 is the "proper SpecFlow v4" release.
+
+Another important and outstanding need that we inherited from SpecFlow was to clean up the execution workflow. Probably the most notable element of that was supporting test-level parallelization, but in the background we also had to redesign the entire resource handling, hook execution, and error handling infrastructure.
+
+## V3 is about SpecFlow ecosystem parity
+
+Users have seen the SpecFlow ecosystem as a homogeneous set of frameworks and supporting tools that consisted of the SpecFlow core framework, the Visual Studio and Rider extensions, the SpecFlow plugins, and the SpecFlow+ toolset — particularly the SpecFlow+ LivingDoc Generator. In reality, each of these parts and tools has its own story, and most importantly, the SpecFlow+ toolset was never open source, so we could not fork it as we did with the other assets.
+
+This has been a pain point for many users, because LivingDoc Generator was the go-to tool for generating nice HTML reports about their specifications and execution results. And we could not take that over.
+
+Although there are alternative solutions for generating HTML reports (notably [Allure Report](https://allurereport.org/), one of the most popular open-source reporting tools, which was one of the first to integrate with Reqnroll), we felt that we needed to provide a built-in solution as well.
+
+Our strategy has been to adopt the concept of "Cucumber Messages" from the Cucumber tool family. Cucumber Messages is basically a standardized file format that can capture all execution details of a BDD framework, including step or hook executions and attachments. Our idea was that if we were able to provide a Cucumber Messages report, we would open up visualization possibilities that have already been available to Cucumber tools. We cannot talk about this feature without mentioning [Chris Rudolphi](https://github.com/clrudolphi), who made an incredible effort to bring this strategy to reality. Thank you, Chris!
```

Comment:
Thank you for including the recognition, but its not necessary. If you must, please generalize it to the entire team. This feature would not have happened without contributions from everyone.